www/search

```diff
@@ -1342,6 +1342,9 @@ else if ($term || $browse)
                     $tagsMatched = array();
                     foreach ($specials as $s) {
                         if ($s[0][0] == "tags") {
+                            // Don't try to match negated terms
+                            if (strpos($s[2], '-') !== false)
```

Comment:
```suggestion
                            if ($s[2] === '-')
```

I think `$s[2]` is necessarily either the one-character string `-` or the empty string.

https://github.com/iftechfoundation/ifdb/blob/a3a6f0b097a49e240ae30769177e9ecdc112da6a/www/searchutil.php#L345-L361

`$s[2]` is `$match[1]`, but the first matching group is `(-?)`, i.e. either a hyphen or the empty string.